unified-codes/apps/data-service/src/app/resolvers.js

```diff
@@ -1,60 +0,0 @@
-// TODO: should make roles an enum?
-const ADMIN_ROLE = 'ADMIN';
-
-const queries = {
-  entity: (code) => {
-    return `{
-        query(func: eq(code, ${code}), first: 1) @recurse(loop: false)  {
-          code
-          description
-          type
-          value
-          has_child
-          has_property
-        }
-      }`;
-  },
-  entities: (type) => {
-    return `{
-      query(func: eq(type, ${type})) @filter(has(description)) @recurse(loop: false)  {
-        code
-        description
-        type
-        value
-        has_child
-        has_property
-      }
-    }`;
-  },
-};
-
-export const resolvers = {
-  Query: {
-    entity: async (_source, _args, context) => {
-      const { dataSources } = context;
-      const { code } = _args;
-      const query = queries.entity(code);
-      const response = await dataSources.dgraph.postQuery(query);
-      const [entity] = response.data.query;
-      return entity;
-    },
-    entities: async (_source, _args, context) => {
-      const { dataSources } = context;
-      const { type = 'medicinal_product' } = _args?.filter ?? {};
-      const query = queries.entities(type);
-      const response = await dataSources.dgraph.postQuery(query);
-      return response.data.query;
-    },
-    user: async (_parent, _args, context) => {
```

Comment:
the `User` endpoint was useful for testing auth. will it reappear somewhere else? or removed for good?

Reply:
Ah, I was a bit confused by the `User` endpoint. My thinking was that the data-service doesn't store anything related to the `User` (it receives a token, but doesn't maintain any state), so I wasn't sure how the frontend was mean't to use this?

At this stage, the idea is the frontend directly grabs the token from keycloak (and then just passes it along with its requests, no graphql types needed). At some stage we might want to add a mutation to `data-service` for the frontend to login (i.e. pass credentials and get the token), but it would remain the case that the `back-end` wouldn't be exposing anything for reading (as it wouldn't be maintaining the user in state, the frontend will just do that with the token). 

Does that make sense or am I missing something?

Reply:
It was just parsing the token and returning the user details in a graph. not exactly useful for anyone - though I thought perhaps it might help with the user avatar. It was useful in testing auth, which was really why I put it in. We don't have mutations yet, so no mutation endpoints to secure.

Now that you've done the auth service changes, that helps anyway.